_includes/headtags.html

```diff
@@ -1,15 +1,18 @@
 <meta charset="utf-8">
-{% for lng in site.languages %}
-    {% if lng == site.default_lang %}
-      {% capture langlink %}{{ site.a_rootpage }}{{ page.url }}{% endcapture %}
-    {% else %}
-      {% capture langlink %}{{ site.a_rootpage }}/{{ lng }}{{ page.url }}{% endcapture %}
-    {% endif %}
-    <link rel="alternate" hreflang="{{ lng }}" href="{{ langlink }}">
+<!-- Canonical and Alternate Links -->
+{% assign sorted_languages = site.languages | sort %}
+{% for lng in sorted_languages %}
+  {% if lng == site.default_lang %}
+    <link rel="canonical" href="{{ site.url }}{{ site.baseurl }}{{ page.url }}">
+  {% else %}
+    <link rel="alternate" hreflang="{{ lng }}" href="{{ site.url }}{{ site.baseurl }}/{{ lng }}{{ page.url }}">
+  {% endif %}
 {% endfor %}
+<link rel="alternate" hreflang="x-default" href="{{ site.url }}/{{ site.default_lang }}{{ page.url }}">
```

Comment:
I believe that this is not correct: it will produce /en/ but there is no /en/ subpage.

```suggestion
<link rel="alternate" hreflang="x-default" href="{{ site.url }}/{{ page.url }}">
```

is probably correct.

Reply:
I hope the whole canonical/alternate thing is correct then.

Reply:
Oh, I see Google says "The new` x-default` hreflang attribute value signals to our algorithms that this page doesn't target any specific language or locale and is the default page when no other page is better suited."

Reply:
Which we could argue is ok for en.

Reply:
> I believe that this is not correct: it will produce /en/ but there is no /en/ subpage.

Yes, you're right.